client/layout/masterbar/logged-in.jsx

```diff
@@ -230,111 +248,186 @@ class MasterbarLoggedIn extends Component {
 					siteSlug={ siteSlug }
 					isLeavingAllowed={ ! isCheckoutPending }
 				/>
-			);
-		}
-
-		return (
-			<>
-				{ isWordPressActionSearchFeatureEnabled && isActionSearchVisible ? (
+			),
+			popupSearch: () =>
+				isWordPressActionSearchFeatureEnabled && isActionSearchVisible ? (
 					<AsyncLoad
 						require="calypso/layout/popup-search"
 						placeholder={ null }
 						onClose={ this.onSearchActionsClose }
 					/>
-				) : null }
+				) : null,
+			mySites: () => this.renderMySites(),
+			reader: () => (
+				<Item
+					tipTarget="reader"
+					className="masterbar__reader"
+					url="/read"
+					icon="reader"
+					onClick={ this.clickReader }
+					isActive={ this.isActive( 'reader' ) }
+					tooltip={ translate( 'Read the blogs and topics you follow' ) }
+					preloadSection={ this.preloadReader }
+				>
+					{ translate( 'Reader', { comment: 'Toolbar, must be shorter than ~12 chars' } ) }
+				</Item>
+			),
+			languageSwitcher: () =>
+				( this.props.isSupportSession || config.isEnabled( 'quick-language-switcher' ) ) && (
+					<AsyncLoad require="./quick-language-switcher" placeholder={ null } />
+				),
+			search: () =>
+				isWordPressActionSearchFeatureEnabled && (
+					<Item
+						tipTarget="Action Search"
+						icon="search"
+						onClick={ this.clickSearchActions }
+						isActive={ false }
+						className="masterbar__item-action-search"
+						tooltip={ translate( 'Search' ) }
+						preloadSection={ this.preloadMe }
+					>
+						{ translate( 'Search Actions' ) }
+					</Item>
+				),
+			planUpsell: () => (
+				<AsyncLoad
+					require="./plan-upsell"
+					className="masterbar__item-upsell button is-primary"
+					tooltip={ translate( 'Upgrade your plan' ) }
+				>
+					{ translate( 'Upgrade' ) }
+				</AsyncLoad>
+			),
```

Comment:
Maybe the planUpsell needs the `! domainOnlySite && ! isMigrationInProgress` check as publish?

Reply:
Phew, great catch.